Lib/test/test_codecs.py

```diff
@@ -3285,18 +3302,26 @@ def check_encode(self, cp, tests):
                     '%a.encode("cp%s", %r)=%a != %a'
                     % (text, cp, errors, encoded[0], expected))
                 self.assertEqual(encoded[1], len(text))
+
+                encoded = text.encode(f'cp{cp}', errors)
+                self.assertEqual(encoded, altexpected,
+                    '%a.encode("cp%s", %r)=%a != %a'
+                    % (text, cp, errors, encoded, altexpected))
             else:
                 self.assertRaises(UnicodeEncodeError,
                     codecs.code_page_encode, cp, text, errors)
+                self.assertRaises(UnicodeEncodeError,
+                    text.encode, f'cp{cp}', errors)
 
     def test_cp932(self):
         self.check_encode(932, (
             ('abc', 'strict', b'abc'),
             ('\uff44\u9a3e', 'strict', b'\x82\x84\xe9\x80'),
+            ('\uf8f3', 'strict', b'\xff'),
             # test error handlers
             ('\xff', 'strict', None),
             ('[\xff]', 'ignore', b'[]'),
-            ('[\xff]', 'replace', b'[y]'),
+            ('[\xff]', 'replace', b'[y]', b'[?]'),
```

Comment:
Why are these updates necessary ? Please clarify.

We should not have two different codec implementations of the same code page returning different values on win32 vs. the rest. If the Windows code page codecs return different data, please make sure that the native ones from the encodings package are used, so that we don't break code.

Reply:
Or put differently: the win32 cp codecs can add new code page support, but may not override existing codecs, if they don't behave in the same way.

Reply:
The first value is what the Windows codec (`codecs.code_page_encode()`) returns. The second is what the Python codec returns.

Previously, this test only tested `codecs.code_page_encode()`. I make it testing also the Python codec, because results are identical in most cases.

The fact that they are different is evidence that the Python codec has preference over the Windows codec.

Reply:
Ok, then that's good. The Windows codecs don't always follow the mappings we use from the Unicode Consortium. Updating the mappings could be discussed in separate issues, but we should continue to use the Unicode Consortium files as basis, not the Microsoft ones.